text/0470-aurora-serverless-v2.md

```diff
@@ -0,0 +1,173 @@
+could be either serverless or privisioned.
+
+```ts
+new rds.DatabaseClusterV2(stack, 'cluster', {
+  engine,
+  vpc,
+  // writer(serverless)
+  writer: { serverless: true },
+  readers: [
+    // reader 1(provisioned)
+    { instanceType: ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.LARGE) },
+    // reader 2(serverless)
+    { serverless: true },
+  ],
+});
+```
+
+> * **PRESS RELEASE**: If this is a major feature (~6 months of work), write the
+>   press release which announces this feature. The press release is a single
+>   page that includes 7 paragraphs: (1) summary, (2) problem, (3) solution, (4)
+>   leader quote, (5) user experience, (6) customer testimonial and (7) one
+>   sentence call to action.
+
+---
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `api-approved` label was applied to the
+RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @xxxxx
+```
+
+## Public FAQ
+
+> This section should include answers to questions readers will likely ask about
+> this release. Similar to the "working backwards", this section should be
+> written in a language as if the feature is now released.
+>
+> The template includes a some common questions, feel free to add any questions
+> that might be relevant to this feature or omit questions that you feel are not
+> applicable.
+
+### What are we launching today?
+
+We are launching the Aurora Serverless v2 support for `aws-rds` that allows users to
+create a new cluster with Serverless v2 support and define serverless or provisioned
+`writer` and `readers` in the new `DatabaseClusterV2` construct.
+
+### Why should I use this feature?
+
+1. I need to create a cluster with Aurora serverless v2 support.
+2. I need to create the writer as well as the readers with this cluster.
+
+## Internal FAQ
+
+> The goal of this section is to help decide if this RFC should be implemented.
+> It should include answers to questions that the team is likely ask. Contrary
+> to the rest of the RFC, answers should be written "from the present" and
+> likely discuss design approach, implementation plans, alternative considered
+> and other considerations that will help decide if this RFC should be
+> implemented.
+
+### Why are we doing this?
+
+The existing `aws-rds` module is missing the Aurora Serverless v2 support and we are adding this support into this module.
+
+### Why should we _not_ do this?
+
+> Is there a way to address this use case with the current product? What are the
+> downsides of implementing this feature?
+
+### What is the technical solution (design) of this feature?
+
+We are creating a new `DatabaseClusterV2` L2 construct by extending the `DatabaseClusterBase` class.
+Users that opt-in serverless v2 enabled clusters should use this class to create a new cluster
+as well as writer and reader instances, which could be serverless, porivioned or mixed.
```

Comment:
How can the user opt-in for this functionality? Is it just by using this construct?

Reply:
Yes. From my current proposed design, the `DatabaseClusterV2` construct is only for serverlessV2-enabled clusters only. When you instantiate this class, a `CfnDatabaseCluster` would be provisioned with serverlessV2 enabled.